### PR TITLE
Unset KUBERNETES_PROVIDER  when KUBERNETES_CONFORMANCE_TEST is set

### LINKED ISSUE #26269: Conformance tests try to use gce as provider, when it should be provider independent
I'm trying to run e2e tests against my k8s cluster

```bash
export E2E_REPORT_DIR=${WORKSPACE}/results/
export NUM_MINIONS=2
export KUBERNETES_CONFORMANCE_TEST=y
export KUBECONFIG=${WORKSPACE}/k8s_files/kubeconfig.yml
hack/ginkgo-e2e.sh -host=https://10.91.97.170
```
 but i got error

```bash
00:00:17.997 /var/lib/jenkins/workspace/kubernetes-e2e-conformance-docker/src/k8s.io/kubernetes/cluster/../cluster/gce/util.sh: line 37: gcloud: command not found
00:00:18.000 !!! Error in /var/lib/jenkins/workspace/kubernetes-e2e-conformance-docker/src/k8s.io/kubernetes/cluster/../cluster/gce/util.sh:37
00:00:18.000   'gci_images=($(gcloud compute images list --project google-containers       --regexp='gci-dev.*' --format='value(name)'))' exited with status 127
00:00:18.001 Call stack:
00:00:18.001   1: /var/lib/jenkins/workspace/kubernetes-e2e-conformance-docker/src/k8s.io/kubernetes/cluster/../cluster/gce/util.sh:37 source(...)
00:00:18.001   2: /var/lib/jenkins/workspace/kubernetes-e2e-conformance-docker/src/k8s.io/kubernetes/cluster/kube-util.sh:100 source(...)
00:00:18.001   3: hack/ginkgo-e2e.sh:44 main(...)
00:00:18.001 Exiting with status 1
```

```KUBERNETES_CONFORMANCE_TEST=y``` should set ```KUBERNETES_PROVIDER``` to "", but its changed somehow to gce

Until yesterday it was working

### PR DESCRIPTION
**Pull request by pskrzyns**
fixes #26269

### CODE REVIEW

File: cluster/kube-util.sh [hunk 1]
```diff
@@ -16,10 +16,16 @@
...
 # This script contains skeletons of helper functions that each provider hosting
 # Kubernetes must implement to use cluster/kube-*.sh scripts.
-# It sets KUBERNETES_PROVIDER to its default value (gce) if it is unset, and
+# It sets KUBERNETES_PROVIDER to its default value (gce) if it is unset
+# and KUBERNETES_CONFORMANCE_TEST is unset, and
 # then sources cluster/${KUBERNETES_PROVIDER}/util.sh.
 
-KUBERNETES_PROVIDER="${KUBERNETES_PROVIDER:-gce}"
+if [[ -n "${KUBERNETES_CONFORMANCE_TEST:-}" ]]; then
+    KUBERNETES_PROVIDER=""
```
Comment (sttts) [inline cluster/kube-util.sh]: It even unsets it if it has been set before (in contrast to the PR subject).

File: cluster/kube-util.sh [hunk 2]
```diff
@@ -16,10 +16,16 @@
...
+else
+    KUBERNETES_PROVIDER="${KUBERNETES_PROVIDER:-gce}"
+fi
+
```
Comment (sttts) [inline cluster/kube-util.sh]: nit: space too much

File: cluster/kube-util.sh (hunk 1 above)
Comment (pskrzyns) [inline cluster/kube-util.sh]: And i think its desired behavior
Its the same as in https://github.com/kubernetes/kubernetes/blob/master/hack/ginkgo-e2e.sh#L47

Comment (sttts) [inline cluster/kube-util.sh]: I don't doubt that, only highlighting that subject and patch differ :-)

Comment (pskrzyns) [inline cluster/kube-util.sh]: Fixed

**Comment by spiffxp** [comment]
:+1: at present conformance tests will fail if gcloud isn't installed and the provider is left unset

/cc @kubernetes/sig-testing

File: cluster/kube-util.sh (hunk 1 above)
Comment (ixdy) [inline cluster/kube-util.sh]: I'm not sure why we do that. if anything, I'd be for removing that line in ginkgo-e2e.sh.

**Comment by mikedanese** [comment]
This looks fine. @k8s-bot ok to test

**Comment by pskrzyns** [comment]
Moved to #28164 cause I accidentally lost my fork

**Pull request closed without merging**